src/services/catalog.service.ts

```diff
@@ -96,25 +96,39 @@ export class CatalogService {
     return new TemplateItem(resource, this);
   }
 
-  public getCategoriesBySubCategories(tags: any) {
+  public getCategoriesBySubCategories(itemTags: any) {
     let catsBySubCats = {};
     let otherId = 'other';
-
-    _.each(tags, (tag) => {
-      _.each(this.categories, (category) => {
-        let subCat: any = _.find(category.subCategories, (subCategory: any) => {
-              return subCategory.id === tag || _.includes(subCategory.categoryAliases, tag);
-            }
-        );
-        if (subCat) {
-          catsBySubCats[subCat.id] = category.id;
-          return false;
+    let filteredSubCats: any;
+
+    _.each(this.categories, (category) => {
+      if (category.tags) {
+        if (!_.isEmpty(this.getMatchingTags(category.tags, itemTags))) {
+          filteredSubCats = this.filterSubCatsByTags(category.subCategories, itemTags);
+          if (!_.isEmpty(filteredSubCats)) {
+            _.each(filteredSubCats, (subCat) => {
+              catsBySubCats[subCat.id] = category.id;
+            });
+          } else {
+            catsBySubCats[otherId] = category.id;
```

Comment:
I don't see how that is possible.  catsBySubCats is a single map and your keys are the subcategory id's, what you are storing as the value is the main category id, not an array of main category ids. So the outer loop is going to initially set `catsBySubCats['other'] = "mycat-a"` and on the the second iteration of the outer loop it is going to set `catsBySubCats['other'] = "mycat-b"` which is overwriting what you set before, not adding to it.